src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Completion/CompletionHandler.cs

```diff
@@ -0,0 +1,217 @@
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Roslyn.Utilities;
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.EditorConfigSettings.Data;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Options;
+using System.Net.WebSockets;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig.Features
+{
+    [ExportStatelessLspService(typeof(CompletionHandler), ProtocolConstants.EditorConfigLanguageContract), Shared]
+    [Method(Methods.TextDocumentCompletionName)]
+    internal class CompletionHandler : IRequestHandler<CompletionParams, CompletionList?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CompletionHandler()
+        {
+        }
+
+        public bool MutatesSolutionState => false;
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(CompletionParams request) => request.TextDocument;
+
+        public async Task<CompletionList?> HandleRequestAsync(CompletionParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.AdditionalDocument;
+            Contract.ThrowIfNull(document);
+
+            var workspace = context.Solution?.Workspace;
+            Contract.ThrowIfNull(workspace);
+
+            Contract.ThrowIfNull(request.Context);
+
+            var filePath = document.FilePath;
+            Contract.ThrowIfNull(filePath);
+
+            var optionSet = workspace.Options;
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var offset = text.Lines.GetPosition(ProtocolConversions.PositionToLinePosition(request.Position));
+            var textInLine = text.Lines.GetLineFromPosition(offset).ToString();
+
+            var settingsAggregator = workspace.Services.GetRequiredService<ISettingsAggregator>();
+
+            var codeStyleProvider = settingsAggregator.GetSettingsProvider<CodeStyleSetting>(filePath);
+            var whitespaceProvider = settingsAggregator.GetSettingsProvider<WhitespaceSetting>(filePath);
+            var analyzerProvider = settingsAggregator.GetSettingsProvider<AnalyzerSetting>(filePath);
+
+            var codeStyleSnapshot = codeStyleProvider?.GetCurrentDataSnapshot();
+            var whitespaceSnapshot = whitespaceProvider?.GetCurrentDataSnapshot();
+            var analyzerSnapshot = analyzerProvider?.GetCurrentDataSnapshot();
+
+            // Check if the user has written a name and not a value
+            if (textInLine.Contains(' ') && !textInLine.Contains('='))
```

Comment:
so this will hit if you type a space on an empty line.  Up for debate, but it seems reasonable to offer completion.  You could even offer a text edit that would remove the extra space as part of commit